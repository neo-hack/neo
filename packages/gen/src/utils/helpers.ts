import { Workflow } from '../interface'
import { hooks } from '../utils/hooks'

import Observable from 'zen-observable'
import type Listr from 'listr'
import { LIFE_CYCLES } from 'src/constants'

/**
 * @description convert schema to listr
 */
export const toListr = (schema: Workflow) => {
  const tasks: Listr.ListrTask[] = Object.keys(schema.jobs || {}).map((key) => {
    const taskName = schema.jobs![key].name || key
    return {
      title: taskName,
      task: () => {
        let ob: ZenObservable.SubscriptionObserver<any>
        const OB = new Observable<any>((observer) => {
          ob = observer
        })
        hooks.addHooks({
          [LIFE_CYCLES.STEP]: (args: { step: string }) => {
            ob.next(args.step)
          },
          [taskName]: () => {
            ob.complete()
          },
        })
        return OB as any
      },
    }
  })
  return tasks
}
