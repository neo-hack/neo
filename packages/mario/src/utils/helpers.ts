import { Workflow } from '../interface'
import { hooks } from './hooks'

import Observable from 'zen-observable'
import type Listr from 'listr'
import { LIFE_CYCLES } from 'src/constants'

/**
 * @description convert schema to listr
 */
export const toListr = (schema: Workflow) => {
  const tasks: Listr.ListrTask[] = Object.keys(schema.jobs || {}).map((key) => {
    const taskName = schema.jobs![key].name || key
    return {
      title: taskName,
      task: () => {
        let ob: ZenObservable.SubscriptionObserver<any>
        const OB = new Observable<any>((observer) => {
          ob = observer
        })
        hooks.addHooks({
          [LIFE_CYCLES.STEP]: (args: { step: string }) => {
            ob.next(args.step)
          },
          [taskName]: (args: { error: Error }) => {
            if (!args.error) {
              ob.complete()
              return
            }
            ob.error(args.error)
          },
        })
        return OB as any
      },
    }
  })
  return tasks
}
