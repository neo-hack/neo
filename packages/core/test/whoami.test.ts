import { execNeo } from './helpers'

describe('command whoami', () => {
  it('should print neo', async () => {
    const { stdout, stderr } = await execNeo(['whoami'])
    expect(stderr).toBe('')
    expect(typeof stdout).toBe('string')
  })
})
