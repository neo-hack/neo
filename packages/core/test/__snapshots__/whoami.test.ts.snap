// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`command whoami should print neo 1`] = `
"    ___       ___       ___   
   /\\\\__\\\\     /\\\\  \\\\     /\\\\  \\\\  
  /:| _|_   /::\\\\  \\\\   /::\\\\  \\\\ 
 /::|/\\\\__\\\\ /::\\\\:\\\\__\\\\ /:/\\\\:\\\\__\\\\
 \\\\/|::/  / \\\\:\\\\:\\\\/  / \\\\:\\\\/:/  /
   |:/  /   \\\\:\\\\/  /   \\\\::/  / 
   \\\\/__/     \\\\/__/     \\\\/__/  

 ⎯⎯
 Homepage (​https://github.com/neo-hack/neo#readme​)"
`;
